### PR TITLE
Add load balancing options

### PR DESCRIPTION
**Pull request by srnagar**
This PR adds new options to customize load balancer for event processor client.
- Load balancer update interval
- Partition ownership expiration interval
- Load balancer strategy to support greedy and balanced modes

Fixes https://github.com/Azure/azure-sdk-for-java/issues/5088

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java [hunk 1]
```diff
@@ -421,9 +481,17 @@ public EventProcessorClient buildEventProcessorClient() {
         }
 
         final TracerProvider tracerProvider = new TracerProvider(ServiceLoader.load(Tracer.class));
+        if (loadBalancingUpdateInterval == null) {
+            loadBalancingUpdateInterval = DEFAULT_LOAD_BALANCING_UPDATE_INTERVAL;
+        }
+        if (partitionOwnershipExpirationInterval == null) {
```
Comment (YijunXieMS) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java:487]: In another `EventProcessorClientBuilder` I see "@throws NullPointerException if {@code partitionOwnershipExpirationInterval} is {@code null}". Could this be Null?

Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java:487]: Yes, the `partitionOwnershipExpirationInterval` is null by default. If the user attempts to set this using `public EventProcessorClientBuilder partitionOwnershipExpirationInterval(Duration partitionOwnershipExpirationInterval)` method, then we check that the user-provided value is not null/negative/zero.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/LoadBalancingStrategy.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+     * towards a stable state where all active processors will have an even distribution of Event Hub partitions.
+     * This strategy may take longer to settle into a balanced partition distribution among active processor
+     * instances. This strategy is geared towards minimizing ownership contention and reducing the need to transfer
+     * ownership frequently, especially when multiple instances are initialized together, until a stable state is
+     * reached.
+     */
+    BALANCED,
+
+    /**
+     * The event processor will attempt to claim it's fair share of partition ownership greedily. This enables event
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/LoadBalancingStrategy.java:23]: it's -> its

**Comment by srnagar** [comment]
Merged to another release branch. Closing this PR.
https://github.com/Azure/azure-sdk-for-java/pull/12932

**Pull request closed without merging**